pkg/pipeline/pipeline_netobs_test.go

```diff
@@ -0,0 +1,201 @@
+package pipeline
+
+import (
+	"fmt"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/api"
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/decode"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/ingest"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/transform/netobserv"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/write"
+	"github.com/netobserv/flowlogs-pipeline/pkg/test"
+	"github.com/prometheus/common/model"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+// Tests in this file are aimed at validating the Flowlogs Pipeline configuration for its
+// usage within the network observability kube enrichment
+
+const timeout = 5000 * time.Second
+
+type testPipeline struct {
+	pipe          *Pipeline
+	client        *test.IPFIXClient
+	lokiFlows     chan map[string]interface{}
+	fakeloki      *httptest.Server
+	stopInformers chan struct{}
+}
+
+func (p *testPipeline) Close() {
+	close(p.lokiFlows)
+	close(p.stopInformers)
+	p.fakeloki.Close()
+}
+
+// NetobsTestPipeline reproduces a basic network observability pipeline and returns it.
+// It is not created from the global YAML text configuration because
+// we need a programmatic way to instantiate a pipeline and inject there the informers
+func NetobsTestPipeline(t *testing.T) *testPipeline {
+	// STAGE 1: IPFIX ingester
+	// Find a free port for the IPFIX collector
+	ipfixPort, err := test.UDPPort()
+	require.NoError(t, err)
+	collector, err := ingest.NewIngestCollector(api.IngestCollector{
+		HostName:    "0.0.0.0",
+		Port:        ipfixPort,
+		BatchMaxLen: 1,
+	})
+	require.NoError(t, err)
+
+	// STAGE 2: decoding from flow's JSON data to GenericMAP
+	decoder, err := decode.NewDecodeJson()
+	require.NoError(t, err)
+
+	// STAGE 3: Kube Enricher getting its data from a mocked informer
```

Comment:
We do not have the term "enrich" as part of FLP ... I think that we might replace transform with `enrich` (if you think that this is correct for kube) but we need to do that across all the project ... 

There are multiple placed in this test file that need to be updated